R/barchart.R

```diff
@@ -463,6 +464,16 @@ Bar <- function(x,
                             if (nzchar(data.label.prefix[ii,i])) list(list(Text = data.label.prefix[ii,i])) else NULL,
                             list(list(Field="Value")),
                             if (nzchar(tmp.suffix[ii])) list(list(Text = tmp.suffix[ii])) else NULL)))
+
+                    if (multi.color.labels)
+                    {
+                        for (ii in 1:nrow(chart.matrix)) 
+                        {
+                            for (j in 1:length(pt.segs[[ii]]$Segments))
```

Comment:
Can `pt.segs[[ii]]$Segments` be `null`?

Reply:
Segments is initialized in line 461, so it can't be null here. We do check afterwards if we can trim it back, but at this stage we are just iterating through the settings and annotations to see if there is anything to be exported